terror_error.go

```diff
@@ -284,10 +278,84 @@ func (e *Error) UnmarshalJSON(data []byte) error {
 	e.line = err.Line
 	e.file = err.File
 	e.message = err.Error
-	e.class = &ErrClass{
-		Description: className,
-		ID:          err.Class,
-		registry:    &Registry{Name: regName},
-	}
 	return nil
 }
+
+func (e *Error) Wrap(err error) *Error {
+	e.cause = err
+	return e
+}
+
+func (e *Error) Cause() error {
+	root := Unwrap(e.cause)
+	if root == nil {
+		return e.cause
+	}
+	return root
+}
+
+func (e *Error) FastGenWithCause(args ...interface{}) error {
+	err := *e
+	err.message = e.cause.Error()
+	err.args = args
+	return SuspendStack(&err)
+}
+
+func (e *Error) GenWithStackByCause(args ...interface{}) error {
+	err := *e
+	err.message = e.cause.Error()
+	err.args = args
+	err.fillLineAndFile(1)
+	return AddStack(&err)
+}
+
+type NormalizeOption func(*Error)
+
+// Description returns a NormalizeOption to set description.
+func Description(desc string) NormalizeOption {
+	return func(e *Error) {
+		e.description = desc
+	}
+}
+
+// Workaround returns a NormalizeOption to set workaround.
+func Workaround(wr string) NormalizeOption {
+	return func(e *Error) {
+		e.workaround = wr
+	}
+}
+
+// RFCCodeText returns a NormalizeOption to set RFC error code.
+func RFCCodeText(codeText string) NormalizeOption {
+	return func(e *Error) {
+		e.codeText = ErrCodeText(codeText)
+	}
+}
+
+// MySQLErrorCode returns a NormalizeOption to set error code.
+func MySQLErrorCode(code int) NormalizeOption {
+	return func(e *Error) {
+		e.code = ErrCode(code)
+	}
+}
+
+// Normalize creates a new Error object.
+func Normalize(message string, opts ...NormalizeOption) *Error {
+	e := &Error{
+		message: message,
+	}
+	for _, opt := range opts {
+		opt(e)
+	}
+	return e
+}
+
+// CauseError returns zap.Field contains cause error.
+func CauseError(err *Error) zap.Field {
+	return zap.Field{Key: "error", Type: zapcore.ErrorType, Interface: err.FastGenWithCause()}
+}
+
+// CauseError returns zap.Field contains error.
```

Comment:
```suggestion
// DetailError returns zap.Field contains error.
```

Reply:
BTW, I think we can combine these two methods? And Wrap the cause error inside of this function so that we can directly use something like `CauseError(err, causeErr)`.